publications/radstack/radstack.tex

```diff
@@ -0,0 +1,1148 @@
+\cite{yang2014druid}, and we’ve found that the combination of technologies is
+flexible enough to handle a wide variety of processing requirements and query
+loads. Each piece of the stack is designed to do a specific set of things very
+well. This paper will cover the details and design principles of the RADStack.
+Our contributions are around the architecture of the stack itself, the
+introduction of Druid as a serving layer, and our model for unifying real-time
+and historical workflows.
+
+The structure of the paper is as follows: Section \ref{sec:background}
+describes the problems and use cases that led to the creation of the RADStack.
+Section \ref{sec:serving} describes Druid, the serving layer of the stack, and
+how Druid is built for real-time and batch data ingestion, as well as
+exploratory analytics. Section \ref{sec:processing} covers the role of Samza
+and Hadoop for data processing, and Section \ref{sec:delivery} describes the
+role of Kafka for event delivery. In Section \ref{sec:performance}, we present
+our production metrics.  Section \ref{sec:experiences} presents our experiences
+with running the RADStack in production, and in Section \ref{sec:related} we
+discuss the related solutions.
+
+\section{Background}
+\label{sec:background}
+
+\begin{table*}
+\centering
+\begin{tabular}{| l | l | l | l | l | l | l | l |}
+  \hline
+  \textbf{Timestamp} & \textbf{Publisher} & \textbf{Advertiser} & \textbf{Gender} & \textbf{City} & \textbf{Click} & \textbf{Price} \\ \hline
+  2011-01-01T01:01:35Z & bieberfever.com & google.com & Male & San Francisco & 0 & 0.65 \\ \hline
+  2011-01-01T01:03:63Z & bieberfever.com & google.com & Male & Waterloo & 0 & 0.62 \\ \hline
+  2011-01-01T01:04:51Z & bieberfever.com & google.com & Male & Calgary & 1 & 0.45 \\ \hline
+  2011-01-01T01:00:00Z & ultratrimfast.com & google.com & Female & Taiyuan & 0 & 0.87 \\ \hline
+  2011-01-01T02:00:00Z & ultratrimfast.com & google.com & Female & New York & 0 & 0.99 \\ \hline
+  2011-01-01T02:00:00Z & ultratrimfast.com & google.com & Female & Vancouver & 1 & 1.53 \\ \hline
+\end{tabular}
+\caption{Sample ad data. These events are created when users views or clicks on ads.}
+\label{tab:sample_data}
+\end{table*}
+
+The development of RADStack came about iteratively. Different technologies were
+introduced to address new problems and use cases that came up over time.
+The initial concept of the RADStack started at Metamarkets, an analytics
+startup focused in the advertising technology space, and has since been adopted
+by other technology companies \cite{2014yahoo}. In the early days of Metamarkets, we wanted
+to build an analytics dashboard where users could arbitrarily explore their
+data. An example of the data found in online advertising space is shown in
+Table~\ref{tab:sample_data}. 
+
+The data shown in Table~\ref{tab:sample_data} is fairly standard in OLAP
+workflows. The data is transactional (timeseries data) and tends to be very
```

Comment:
I sometimes fear that "transactional data" can get confused for "ACID-compliant store".  What if we say "The data is an event stream with a timestamp..."?